spec/cli/test/builder-cache.spec.js

```diff
@@ -0,0 +1,114 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+// Builder syntax tests
+
+'use strict';
+
+require('jasmine-expect');
+const fs = require('fs');
+const run = require('./run');
+
+describe('TestCommand test suite for Builder cache scenario', () => {
+  let commandOut = '',
+    commandSuccess = true;
+
+  beforeEach(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;
+  });
+
+  // reset previous state
+  var removeCacheDir = function(path) {
+    if (fs.existsSync(path)) {
+      fs.readdirSync(path).forEach(function(file, index){
+        var cachedLib = path + "/" + file;
+        if (fs.lstatSync(cachedLib).isDirectory()) { // unexpected case
+          removeCacheDir(cachedLib);
+        } else // remove lib
+          fs.unlinkSync(cachedLib);
+      });
+      fs.rmdirSync(path);
+    }
+  };
+  removeCacheDir(".builder-cache");
+
+  it('should run a command', (done) => {
+    run({
+      configPath:  '/fixtures/builder-cache/.imptest',
+    }).then((res) => {
+      commandSuccess = res.success;
+      commandOut = res.out;
+      done();
+    });
+  });
+
+  it('should verify the output', (done) => {
+    // todo: insert more checks here
+    expect(commandSuccess).toBe(true);
+    expect(commandOut).not.toBeEmptyString();
+    expect(commandOut).toMatch(/Testing succeeded\n/);
+    expect(fs.existsSync(".builder-cache")).toBe(false);
+    done();
+  });
+
+  it('should run a command', (done) => {
+    run({
+      configPath:  '/fixtures/builder-cache/.imptest_cache_true',
+      builderCache: false,
+    }).then((res) => {
+      commandSuccess = res.success;
+      commandOut = res.out;
+      done();
+    });
+  });
+
+  it('should verify the output', (done) => {
+    // todo: insert more checks here
+    expect(commandSuccess).toBe(true);
+    expect(commandOut).not.toBeEmptyString();
+    expect(commandOut).toMatch(/Testing succeeded\n/);
+    expect(fs.existsSync(".builder-cache")).toBe(false);
+    done();
+  });
+
+  it('should run a command', (done) => {
+    run({
+      configPath:  '/fixtures/builder-cache/.imptest_cache_false',
+      builderCache: true,
+    }).then((res) => {
+      commandSuccess = res.success;
+      commandOut = res.out;
+      done();
+    });
+  });
+
+  it('should verify the output', (done) => {
+    // todo: insert more checks here
+    expect(commandSuccess).toBe(true);
+    expect(commandOut).not.toBeEmptyString();
+    expect(commandOut).toMatch(/Testing succeeded\n/);
+    expect(fs.existsSync(".builder-cache")).toBe(true);
```

Comment:
probably need to clean up after tests complete

Reply:
added before and after tests